canvas_sdk/data/patient.py

```diff
@@ -0,0 +1,27 @@
+from datetime import datetime
+from typing import Self
+
+from pydantic import BaseModel as Model
+
+from . import data_access_layer_client as dal_client
+
+
+# TODO: Sync this model with Joe's PR
+class Patient(Model):
+    """Patient model."""
+
+    id: str | None
+    first_name: str
+    last_name: str
+    birth_date: datetime
+
+    @classmethod
+    def get(cls, id: str) -> Self:
+        """Given an ID, get the Patient from the Data Access Layer."""
+        patient = dal_client.get_patient(id)
+        return cls(
+            id=patient.id,
+            first_name=patient.first_name,
+            last_name=patient.last_name,
+            birth_date=patient.birth_date,
+        )
```

Comment:
@aduane and I had a very long, detailed, and productive discussion about how objects should be instantiated and how data items in objects should be populated from the Data Access Layer. We considered a lot of different options, none perfect, but landed on this:

* Use the new class method `get` to explicitly state that you want to fetch the data from the database.
* Use the `__init__` method on the Pydantic class to construct a new one in memory.

There are some downsides downsides, and the design on these features will evolve, but this offers something very simple and usable that will provide customers with new capabilities.